### PR TITLE
feature: Remove dashboard and link user directly to manage add on page

### PR DESCRIPTION
**Pull request by kingisaac95**
## JIRA TICKET NAME:
AOM-37: [Remove dashboard and link user directly to manage add on page](https://issues.openmrs.org/browse/AOM-37)
### SUMMARY:
- Removes the dashboard page open up the Manager Add Ons directly
- Create a Settings Icon in the right side corner and link it to Manage Settings page. So if anyone wants setting then they will easily click that button and will change that.

### CODE REVIEW

**Comment by kingisaac95** [comment]
Please hold on so I fix this failing test.

File: app/css/openmrs-addonmanager.css [hunk 1]
```diff
@@ -136,23 +136,31 @@ a:hover {
...
+    margin: 0 30px;
 }
 
-.home-page-body{
+.home-page-body {
     display: flex;
     flex-direction: row;
     position: relative;
-    height: 150px;
+    /* height: 150px; */
```
Comment (jomadoye) [inline app/css/openmrs-addonmanager.css]: Please let us avoid commented out code.

File: app/js/components/manageApps/deleteAddonModal.jsx [hunk 2]
```diff
@@ -23,8 +23,8 @@ export default class DeleteModal extends Component {
           <p>Are you sure you want to delete <strong>{app.name}</strong>?</p>
         </ModalBody>
         <ModalFooter>
-          <button className='btn btn-default' onClick={hideModal}>Cancel</button>
-          <button type="button" className="btn btn-danger" data-dismiss="modal" onClick={() => handleDelete(app.name)}>Delete</button>
+          <button className="btn btn-default" onClick={hideModal}>Cancel</button>
+          <button type="button" className="btn btn-danger" data-dismiss="modal" id="cancel-btn" onClick={() => handleDelete(app.name)}>Delete</button>
```
Comment (jomadoye) [inline app/js/components/manageApps/deleteAddonModal.jsx]: This Line is exceeding the 180 char limit, let us try to break it into more readable pieces.

File: app/css/openmrs-addonmanager.css (hunk 1 above)
Comment (kingisaac95) [reply app/css/openmrs-addonmanager.css]: I forgot that, but I've already removed it. Thanks for noting that.

**Comment by kingisaac95** [comment]
I'll be going offline. I'll be back in a few hours to fix the failing test @dkayiwa

**Comment by kingisaac95** [comment]
@andela-jomadoye @paulupendo @wanjikum Please do your reviews now. All fixes done

File: app/js/components/manageApps/AddonList.jsx [hunk 3]
```diff
@@ -39,15 +39,14 @@ export const AddonList = ({handleDownload, install, appList, openPage, openModal
                 id="delete-icon-wrapper"
               >
                 {install == false ? 
-                  <i className="glyphicon glyphicon-trash text-danger delete-icon"  onClick={openModal(app)}/> :
-                  <i className="glyphicon glyphicon-download-alt text-primary install-icon"  onClick={(e) => handleDownload(e)}/> 
+                  <i className="glyphicon glyphicon-trash text-danger delete-icon" id="icon-btn" onClick={openModal(app)}/> :
+                  <i className="glyphicon glyphicon-download-alt text-primary install-icon" id="icon-btn" onClick={(e) => handleDownload(e)}/>
```
Comment (jomadoye) [inline app/js/components/manageApps/AddonList.jsx]: Can we break this line into new lines as it exceeds the 180 limit per line.

File: app/js/components/manageApps/ManageApps.jsx [hunk 4]
```diff
@@ -289,67 +292,85 @@ export default class ManageApps extends React.Component {
...
+                    placeholder="Search for an add on.."/>
+                </div>
+                <table className="table table-bordered table-striped table-hover">
+                  <thead>
+                    <tr>
+                      <th>Logo</th>
+                      <th>Name</th>
+                      <th>Developer</th>
+                      <th>Version</th>
+                      {this.state.install == false ?
```
Comment (jomadoye) [inline app/js/components/manageApps/ManageApps.jsx]: Can we use the `Identity` operator here (===) as opposed to using the `equality` operator (==)?

File: app/js/Routes.jsx [hunk 5]
```diff
@@ -8,18 +8,16 @@
  */
 import React, { Component } from 'react';
 import { Router, Route, Link, IndexRoute, hashHistory, browserHistory } from 'react-router';
-import App from './components/App';
-import HomePage from './components/homePage/HomePage';
-import ManageApps from './components/homePage/manageApps/ManageApps';
-import ManageSettings from './components/homePage/ManageSettings';
+import App from './components/App.jsx';
```
Comment (jomadoye) [inline app/js/Routes.jsx:11]: Currently across the codebase, we have a trend of importing modules without the extension, I would recommend we use a `consistent import style`. If you are going to import modules with their extensions I think you should ensure all other import statements in the codebase have extensions and vice versa.

File: app/js/Routes.jsx [hunk 6]
```diff
@@ -8,18 +8,16 @@
...
 
 export default () => {
   return (
     <Router history={hashHistory}>
       <Route path="/" component={App} >
-        <IndexRoute component={HomePage}/>
+        <IndexRoute component={ManageApps}/>
         <Route path="manageSettings" component={ManageSettings} />
-        <Route path="manageApps" component={ManageApps} />
       </Route>
```
Comment (jomadoye) [inline app/js/Routes.jsx]: I think we should have a fallback route for when the user tries to access an incorrect URL, I would recommend using the `ManageApp` page as the fallback route.
Adding `<Route path="*" component ={ManageApps}/>` should solve this.

File: app/js/Routes.jsx (hunk 5 above)
Comment (kingisaac95) [reply app/js/Routes.jsx:11]: There's no such trend @andela-jomadoye
If you take a closer look, you'll see that I made it consistent across the whole codebase.
Thank you

File: app/js/Routes.jsx (hunk 6 above)
Comment (kingisaac95) [reply app/js/Routes.jsx]: That was not originally in the plan but it sure sounds good.

File: app/js/components/manageApps/ManageApps.jsx (hunk 4 above)
Comment (kingisaac95) [reply app/js/components/manageApps/ManageApps.jsx]: That piece of code was not written by me, it was part of the changes I pulled after the other PRs were merged, but I'm sure gonna fix it.

File: app/js/components/manageApps/AddonList.jsx (hunk 3 above)
Comment (kingisaac95) [reply app/js/components/manageApps/AddonList.jsx]: This was an oversight, thanks for pointing it out.

**Comment by wanjikum** [comment]
Good work @kingisaac95

**Comment by jomadoye** [comment]
Good work @kingisaac95

File: app/js/Routes.jsx (hunk 5 above)
Comment (jomadoye) [reply app/js/Routes.jsx:11]: Noted @kingisaac95.

File: app/js/components/App.jsx [hunk 7]
```diff
@@ -7,7 +7,7 @@
  * graphic logo is a trademark of OpenMRS Inc.
  */
 import React, { Component } from 'react';
-import HomePage from '../components/homePage/HomePage';
+import ManageApps from '../components/manageApps/ManageApps.jsx';
 import Header from '../components/common/Header';
```
Comment (jomadoye) [inline app/js/components/App.jsx:11]: You may want to add a file extension for this line.

File: app/js/components/manageApps/ManageApps.jsx [hunk 8]
```diff
@@ -8,11 +8,14 @@
  */
 import React from 'react';
 import axios from 'axios';
+import { Router, Route, Link, IndexRoute, hashHistory, browserHistory } from 'react-router';
 import AddAddon from '../manageApps/AddAddon.jsx';
-import { ApiHelper } from '../../../helpers/apiHelper';
-import { AddonList } from './AddonList';
+import BreadCrumbComponent from '../breadCrumb/BreadCrumbComponent.jsx';
+import Header from '../common/Header';
```
Comment (jomadoye) [inline app/js/components/manageApps/ManageApps.jsx:14]: You may want to add a file extension for this line.

File: tests/components/App.test.js [hunk 9]
```diff
@@ -12,10 +12,8 @@ import React from 'react';
 import { mount, shallow } from 'enzyme';
 import { expect } from 'chai';
 import sinon from 'sinon';
-import App from '../../app/js/components/App';
+import App from '../../app/js/components/App.jsx';
 import Header from '../../app/js/components/common/Header';
```
Comment (jomadoye) [inline tests/components/App.test.js:16]: Ditto.

File: app/js/components/manageApps/ManageApps.jsx [hunk 10]
```diff
@@ -289,67 +292,85 @@ export default class ManageApps extends React.Component {
...
+                  <i className="glyphicon glyphicon-cog settings-icon" id="icon-btn" />
+                </Link>
+              </div>
+            </div>
+          </div>
+          
+          <div className="home-page-body">
+            <div className="container-fluid">
+              <div id="notification-wrapper">
+                {this.state.showMsg ? alert : null}
```
Comment (jomadoye) [inline app/js/components/manageApps/ManageApps.jsx]: I believe you did not write this piece of code, however I would like to suggest we `short circuit` this line as opposed to rendering `null`. I.e `{this.state.showMsg && alert}`.
This is inline with ReactJS conditional rendering best pratices [HERE](https://facebook.github.io/react/docs/conditional-rendering.html).  What do you think?

File: app/js/components/manageApps/AddonList.jsx [hunk 11]
```diff
@@ -38,16 +38,23 @@ export const AddonList = ({handleDownload, install, appList, openPage, openModal
                 className="text-center"
                 id="delete-icon-wrapper"
               >
-                {install == false ? 
-                  <i className="glyphicon glyphicon-trash text-danger delete-icon"  onClick={openModal(app)}/> :
-                  <i className="glyphicon glyphicon-download-alt text-primary install-icon"  onClick={(e) => handleDownload(e)}/> 
+                {
+                  install == false ?
```
Comment (jomadoye) [inline app/js/components/manageApps/AddonList.jsx]: I believe you did not write this piece of code, however I would like to suggest we use `strict comparison` i.e (===), as opposed to using `type–converting comparisons`(==) here. What do you think?

File: app/js/components/App.jsx (hunk 7 above)
Comment (kingisaac95) [reply app/js/components/App.jsx:11]: The file is a .js file so there's no need to add the extension. I added the extensions for all .jsx files

File: app/js/components/manageApps/ManageApps.jsx (hunk 8 above)
Comment (kingisaac95) [reply app/js/components/manageApps/ManageApps.jsx:14]: The file is a .js file so there's no need to add the extension. I added the extensions for all .jsx files

File: tests/components/App.test.js (hunk 9 above)
Comment (kingisaac95) [reply tests/components/App.test.js:16]: The file is a .js file so there's no need to add the extension. I added the extensions for all .jsx files

File: app/js/components/manageApps/AddonList.jsx (hunk 11 above)
Comment (kingisaac95) [reply app/js/components/manageApps/AddonList.jsx]: Sure thing

File: app/js/components/App.jsx (hunk 7 above)
Comment (jomadoye) [reply app/js/components/App.jsx:11]: The header is a .jsx component, I understand you didn’t create it, but let us be consistent in our naming, we can’t call some component .js and some .jsx. We need to be consistent whichever one we use. Thank you.

Comment (kingisaac95) [reply app/js/components/App.jsx:11]: What I've done is to change the file into a `.jsx` and then change the import. That will do.

File: app/js/components/manageApps/ManageApps.jsx (hunk 8 above)
Comment (kingisaac95) [reply app/js/components/manageApps/ManageApps.jsx:14]: Ditto

File: tests/components/App.test.js (hunk 9 above)
Comment (kingisaac95) [reply tests/components/App.test.js:16]: Ditto

**Pull request merged**